openapi.yaml

```diff
@@ -132,147 +502,323 @@ securityDefinitions:
     scopes:
       full: Full access to the API
 definitions:
-  Meta:
+  PizzasResults:
     properties:
-      totalResults:
-        type: integer
-        description: Total number of results
-        example: 10
-      totalPages:
-        type: integer
-        description: Total number of pages
-        example: 10
-      currentPageNumber:
-        type: integer
-        description: Page number of the returned results
-        example: 1
-      currentPageSize:
-        type: integer
-        description: Number of results per page
-        example: 25
-  SelfLink:
+      data:
+        type: array
+        items:
+          $ref: "#/definitions/PizzaResource"
+  PizzaResult:
     properties:
-      self:
+      data:
+        $ref: "#/definitions/PizzaResource"
+
+  PizzaResource:
+    properties:
+      type: &pizzaType
         type: string
-        format: url
-        description: Self-link of current resource
-  PaginationLinks:
+        description: the type of the resource (pizza)
+        example: pizza
+      id: &pizzaId
+        type: string
+        description: a unique ID of a pizza
+        example: abc123
+      links:
+        $ref: "#/definitions/SelfLink"
+      attributes:
+        $ref: "#/definitions/PizzaAttributes"
+      relationships: &pizzaRelationships
+        properties:
+          dough:
+            properties:
+              data:
+                $ref: "#/definitions/RelationshipResource"
+          ingredients:
+            properties:
+              links:
+                $ref: "#/definitions/SelfLink"
+              data:
+                type: array
+                description: a list of relationship resources with the ingredient type
+                items:
+                  $ref: "#/definitions/RelationshipResource"
+      included: &pizzaIncluded
+        type: array
+        items:
+          oneOf:
```

Comment:
this creates an error in the swagger editor _but_ still produces the right output (a heterogeneous list of doughs and ingredients, see [here](https://jsonapi.org/format/#document-compound-documents))